### PR TITLE
feat: add hide timer option

### PR DESCRIPTION
**Pull request by solaolateju**
Hide timer option added to setting. 
image

Toggle play and pause fixed
image

Zero hour visibility(count up and down) fixed

### CODE REVIEW

File: src/components/modes/student/StudentView.js [hunk 1]
```diff
@@ -95,9 +95,19 @@ export const StudentView = props => {
...
+          );
+
           return (
             <div className={classes.root}>
-              <Grid className={classes.gridRow} container spacing={3}>
+              <Grid
+                className={classes.gridRow}
+                container
+                spacing={3}
+                style={{ visibility: !timerVisible ? 'visible' : 'hidden' }}
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentView.js:109]: Maybe add this to the `div` above?

File: src/components/modes/student/StudentView.js [hunk 2]
```diff
@@ -128,29 +138,12 @@ export const StudentView = props => {
                       aria-label="Start"
                       color="primary"
                       onClick={() => {
-                        start();
+                        if (started) pause();
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentView.js:141]: Always use `{}` in your if/else clauses.

File: src/components/modes/teacher/Settings.js [hunk 3]
```diff
@@ -118,6 +121,11 @@ class Settings extends Component {
     this.setState({ timeControlsVisible: !timeControlsVisible });
   };
 
+  handleChangeTimerVisibility = () => {
+    const { timerVisible } = this.state;
+    this.setState({ timerVisible: !timerVisible });
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:126]: In here you should have a clause where if `timerVisible` (i.e. if you're going to change `timerVisible` to `false`, you should make sure that `headerVisible` is `false`.

Comment (solaolateju) [reply src/components/modes/teacher/Settings.js]: When `timerVisible` is `false`, 
- both header & timer no longer visible.  
- `headerVisible` is turned off and control disabled
- `startImmediately` is turned on and control disabled
- `timeControlsVisible`  removed from the setting.

Screenshot 2020-05-28 at 12 19 22

File: src/components/modes/student/StudentView.js [hunk 4]
```diff
@@ -95,8 +94,16 @@ export const StudentView = props => {
...
           }
+          const TIME_ZERO = 0;
+          const currentHour = Math.floor(
+            getTime() / ONE_MINUTE_IN_MILLIS / ONE_HOUR_IN_MINS,
+          );
+
           return (
-            <div className={classes.root}>
+            <div
+              style={{ visibility: !timerVisible ? 'visible' : 'hidden' }}
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentView.js:104]: I think you have the condition inverse here. If timer is not visible, then it's visible.

File: src/components/modes/teacher/Settings.js [hunk 5]
```diff
@@ -113,9 +113,17 @@ class Settings extends Component {
...
-    const { timeControlsVisible } = this.state;
-    this.setState({ timeControlsVisible: !timeControlsVisible });
+  handleChangeTimerVisibility = () => {
+    const { timerVisible } = this.state;
+    this.setState({ timerVisible: !timerVisible }, () => {
+      if (timerVisible === false) {
+        const settingsToChange = {
+          headerVisible: false,
+        };
+        this.setState({ startImmediately: true });
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:123]: This state change you can do it with the one in line 118.

File: src/components/modes/teacher/Settings.js [hunk 6]
```diff
@@ -113,9 +113,17 @@ class Settings extends Component {
     this.saveSettings(settingsToChange);
   };
 
-  handleChangeTimeControlVisibility = () => {
-    const { timeControlsVisible } = this.state;
-    this.setState({ timeControlsVisible: !timeControlsVisible });
+  handleChangeTimerVisibility = () => {
+    const { timerVisible } = this.state;
+    this.setState({ timerVisible: !timerVisible }, () => {
+      if (timerVisible === false) {
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:119]: Are you sure `timerVisible` has already changed to `false`? You don't have to get it again from state?

File: src/components/modes/teacher/Settings.js [hunk 7]
```diff
@@ -189,35 +197,43 @@ class Settings extends Component {
       <>
         <FormControlLabel
           className={classes.formControl}
+          disabled={timerVisible}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:200]: This should be if `!timerVisible`, right?

File: src/components/modes/teacher/Settings.js (hunk 6 above)
Comment (solaolateju) [reply src/components/modes/teacher/Settings.js]: Thank you @juancarlosfarah. I was getting `timerVisible` from the local state earlier. Fixed now.

**Pull request merged**